test/integration/users_edit_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'test_helper'
+
+class UsersEditTest < ActionDispatch::IntegrationTest
+  include Capybara::DSL
+
+  test "unsuccessful edit" do
+    user = create(:user)
+    get edit_user_path(user)
+    assert_template 'users/edit'
+    patch user_path(user), user: { username: "",
+                                   email: "foo@invalid",
+                                   password:              "foo",
+                                   password_confirmation: "bar" }
+    refute flash.empty?
+    assert_template 'users/edit'
+  end
+
+  test "successful edit" do
+    user = create(:user)
+
+    get edit_user_path(user)
+    assert_template 'users/edit'
+    username = "Jwan622"
+    first_name  = "Jeffrey"
+    last_name  = "Jeffrey"
+    email  = "Jwan622@yahoo.com"
+    city  = "Staten Island"
+    state  = "New York"
+    zipcode = 10305
+    street = "31 Hillwood Court"
+    country = "USA"
+    credit_card_info = 4141414141414141
+    patch user_path(user), { user: { username: username,
+                                   first_name: first_name,
+                                   last_name: last_name,
+                                   email:   email,
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Unnecessary spacing detected.